File: src/main/java/com/harleylizard/magic_things/common/MagicThingsItems.java
package com.harleylizard.magic_things.common;

import net.minecraft.world.item.BlockItem;
import net.minecraft.world.item.Item;
import net.minecraft.world.level.block.Block;

public final class MagicThingsItems {

    public static void register() {

    }

    public static void register(String name, Item item) {

    }

    public static Item blockItem(Block block) {
        return new BlockItem(block, new Item.Properties());
    }

}

File: src/main/kotlin/com/harleylizard/magic_things/common/MagicThingsItems.kt
package com.harleylizard.magic_things.common

import com.harleylizard.magic_things.common.MagicThings.Companion.resourceLocation
import net.minecraft.core.Registry
import net.minecraft.core.registries.BuiltInRegistries
import net.minecraft.world.item.Item

class MagicThingsItems {

    fun register() {

    }

    fun register(name: String, item: Item) {
        Registry.register(BuiltInRegistries.ITEM, name.resourceLocation, item)
    }

}